main/chessy.c

```diff
@@ -140,24 +298,39 @@ void add_move(Move_t move)
     board[move.start.x][move.start.y] = ' ';
 }
 
-void setup_console()
-{
-    // Setup console
-    // TODO: add a setup for UART as well based on kconfig
-    esp_console_repl_t *repl = NULL;
-    esp_console_repl_config_t repl_config = ESP_CONSOLE_REPL_CONFIG_DEFAULT();
-    esp_console_dev_usb_serial_jtag_config_t usbjtag_config = ESP_CONSOLE_DEV_USB_SERIAL_JTAG_CONFIG_DEFAULT();
-    ESP_ERROR_CHECK(esp_console_new_repl_usb_serial_jtag(&usbjtag_config, &repl_config, &repl));
-}
-
 int app_main(int argc, char *argv[])
 {
-    setup_console();
-    init_board();
-    print_board();
+    // Initialize hardware
+    hall_init();
+    configure_led();
+    
+    // Initialize game
+    char board[8][8];
+    init_board(board);
+    print_board(board);
+
+    // Wait for initial board setup
+    uint8_t matrix[ROW_NUM][COL_NUM];
+    bool board_ready = false;
```

Comment:
"board" and "matrix" are terrible names here. At least change the matrix to hall_sensors, hall_matrix, hall or something that will tell you that the matrix is hall sensors.